x-pack/plugins/ml/public/jobs/new_job/advanced/new_job_controller.js

```diff
@@ -651,6 +674,69 @@ module.controller('MlNewJob',
       }
     };
 
+    function runValidateCardinality() {
+      const { STATUS } = $scope.ui.cardinalityValidator;
+      $scope.ui.cardinalityValidator.status = $scope.ui.cardinalityValidator.STATUS.RUNNING;
+
+      const tempJob = mlJobService.cloneJob($scope.job);
+      _.merge(tempJob, getMinimalValidJob());
+
+      ml.validateCardinality(tempJob)
+        .then((response) => {
+          const validationResult = checkCardinalitySuccess(response);
+
+          if (validationResult.success === true) {
+            $scope.ui.cardinalityValidator.status = STATUS.FINISHED;
+            $scope.ui.cardinalityValidator.message = '';
+          } else {
+            $scope.ui.cardinalityValidator.message = `Creating model plots is resource intensive and not recommended
+                  where the cardinality of the selected fields is greater than 100. Estimated cardinality
+                  for this job is ${validationResult.highCardinality}.
+                  If you enable model plot with this configuration we recommend you use a dedicated results index.`;
```

Comment:
Probably worth adding where that dedicated results index is configured. Something like `If you enable model plot with this configuration we recommend you select a dedicated results index on the Job Details tab.` ?

Reply:
Do you mean as a warning callout as well?